src/indicators/drought/chirpsgefs_rainfallforecast.py

```diff
@@ -0,0 +1,227 @@
+    """
+    df_rain = pd.read_csv(rainy_season_path, parse_dates=[onset_col, cessation_col])
+
+    # remove entries where there is no onset and no cessation date, i.e. no rainy season
+    df_rain = df_rain[
+        (df_rain.onset_date.notnull()) | (df_rain.cessation_date.notnull())
+    ]
+    # if onset date or cessation date is missing, set it to Nov 1/Jul1 to make sure all data of that year is downloaded. This happens if e.g. the rainy season hasn't ended yet
+    df_rain[df_rain.onset_date.isnull()] = df_rain[df_rain.onset_date.isnull()].assign(
+        onset_date=lambda df: pd.to_datetime(f"{df.season_approx.values[0]}-11-01")
+    )
+    df_rain[df_rain.cessation_date.isnull()] = df_rain[
+        df_rain.cessation_date.isnull()
+    ].assign(
+        cessation_date=lambda df: pd.to_datetime(
+            f"{df.season_approx.values[0]+1}-07-01"
+        )
+    )
+
+    # get min onset and max cessation for each season across all admins
+    df_rain_seas = df_rain.groupby("season_approx", as_index=False).agg(
+        {onset_col: np.min, cessation_col: np.max}
+    )
+
+    # create a daterange index including all dates within all rainy seasons
+    all_dates = pd.Index([])
+    for i in df_rain_seas.season_approx.unique():
+        seas_range = pd.date_range(
+            df_rain_seas[df_rain_seas.season_approx == i].onset_date.values[0],
+            df_rain_seas[df_rain_seas.season_approx == i].cessation_date.values[0],
+        )
+        all_dates = all_dates.union(seas_range)
+    return all_dates
+
+
+def download_chirpsgefs(date, config, days_ahead, use_cache=True):
+    """
+    Download the chirps-gefs africa forecast for the given year and day of the year
+    We are focussing on the Africa data, since global data gets massive. Nevertheless, even for Africa it gets massive.
+    Note that a part of the data for 2020 is missing due to a change in model
+    days: number of days forecast predicts ahead. When using 15 day, set to empty string
+    """
+
+    date_str = date.strftime("%Y%m%d")
+    chirpsgefs_filepath = (
+        config.CHIRPSGEFS_RAW_DIR
+        / config.CHIRPSGEFS_RAW_FILENAME.format(days_ahead=days_ahead, date=date_str)
+    )
+    if use_cache and chirpsgefs_filepath.exists():
+        logger.debug(
+            f"{chirpsgefs_filepath} already exists and cache is set to True, skipping"
+        )
+        return chirpsgefs_filepath
+    Path(config.CHIRPSGEFS_RAW_DIR).mkdir(parents=True, exist_ok=True)
+    try:
+        year = date.year
+        day_year = str(date.timetuple().tm_yday).zfill(3)
+        # start_day is the day of the year for which the forecast starts
+        download_ftp(
+            config.CHIRPSGEFS_FTP_URL_AFRICA.format(
+                days_ahead=str(days_ahead).zfill(2), year=year, start_day=day_year
+            ),
+            chirpsgefs_filepath,
+        )
+        logger.debug(f"...successfully downloaded {chirpsgefs_filepath}")
+    except Exception as e:
```

Comment:
It's better to use the specific exception, I think here would be `urllib.error.HTTPError`. You could even check the error string and check that it matches the "HTTP Error 404", so that you don't accidentally miss other http errors